kubernetes/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterScheduler.scala

```diff
@@ -80,14 +80,32 @@ private[spark] class KubernetesClusterScheduler(conf: SparkConf)
 
     // This is the URL of the client jar.
     val clientJarUri = args.userJar
-    conf.setExecutorEnv("spark.executor.jar", clientJarUri)
-    conf.setExecutorEnv("spark.kubernetes.namespace", nameSpace)
-    conf.setExecutorEnv("spark.kubernetes.driver.image", sparkDriverImage)
 
     // This is the kubernetes master we're launching on.
     val kubernetesHost = "k8s://" + client.getMasterUrl().getHost()
     logInfo("Using as kubernetes-master: " + kubernetesHost.toString())
 
+    val submitArgs = scala.collection.mutable.ArrayBuffer.empty[String]
+    submitArgs ++= Vector(
+      clientJarUri,
+      s"--class=${args.userClass}",
+      s"--master=$kubernetesHost",
+      s"--executor-memory=${driverDescription.mem}",
```

Comment:
Note to self: another to-do is to make sure the executor pods are configured with core/mem requirements that align with the corresponding spark configurations for core/mem